kissnet.hpp

```diff
@@ -567,6 +578,10 @@ namespace kissnet
 
 		///operatic-system type for a socket object
 		SOCKET sock;
+#ifdef KISSNET_USE_OPENSSL
+		SSL *ssl;
```

Comment:
Are these pointer taking ownership of the SSL and SSL_CTX objects here? If so, they **need** to be properly handled in the move constructor and the destructor of the class.

Best would be to have them as std::unique_ptr<>. But at the very least we need to make sure to copy and reset these pointers in move operations of the socket class (i.e. `socket(socket&&)` and `socket& operator=(socket&&)`

Reply:
I didn't figured how to move them properly in classes and etc, but will look at this little bit later.

Reply:
The gist of it is, in the move constructor/move assignment operator you want 2 things to occur : 

1) The "moved to" object get the value of these pointers
2) The "moved from" object get these pointers set back to `nullptr`

Move semantics in C++ are recent (C++11 and up) and are genuinely a bit confusing, especially the fact that they use these "r-value references" (you know `type&&`).

I know many C++ programmer that learned the old school stuff and did not refresh to the new standards thoroughly struggle with that specific thing. If you want to read more about it you should google for "C++11 move semantics"

Reply:
I've did research and seems like unique_ptr are not required for SSL pointers. I can be wrong but its what I got.

Did std::move for SSL pointers and better way initialization of OpenSSL.

Reply:
I'll take a look but `std::move()` should not be required. A pointer's just a number.

Reply:
Actually, I used std::move everywhere there... It's technically not wrong so whatever. I'll fix that later.

I need to //TODO some automated testing for this kind of stuff to be honest, but that's a subject *for another day* 🤔

Reply:
Soon I'm going to learn about catch library and can try to integrate it in kissnet if need.